src/Raven.Server/ServerWide/Commands/ETL/AddEtlCommand.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using Raven.Client.Server;
+using Raven.Client.Server.ETL;
+using Raven.Server.Documents.ETL;
+using Raven.Server.Utils;
+using Sparrow.Json.Parsing;
+
+namespace Raven.Server.ServerWide.Commands.ETL
+{
+    public abstract class AddEtlCommand<T> : UpdateDatabaseCommand where T : EtlDestination
+    {
+        public readonly EtlConfiguration<T> Configuration;
+
+        protected AddEtlCommand() : base(null)
+        {
+            // for deserialization
+        }
+
+        protected AddEtlCommand(EtlConfiguration<T> configuration, string databaseName) : base(databaseName)
+        {
+            Configuration = configuration;
+        }
+
+        protected void Add(List<EtlConfiguration<T>> etls, EtlConfiguration<T> configuration)
+        {
+            if (etls == null)
```

Comment:
Huh? You are creating a new instance here, but this is on a local variable.
Did you mean to use a `ref` here?